src/anomalib/data/utils/generators/perlin.py

```diff
@@ -1,160 +1,317 @@
-"""Helper functions for generating Perlin noise."""
-
-# Original Code
-# Copyright (c) 2021 VitjanZ
-# https://github.com/VitjanZ/DRAEM.
-# SPDX-License-Identifier: MIT
-#
-# Modified
+"""Perlin noise-based synthetic anomaly generator."""
+
 # Copyright (C) 2022-2024 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
 
-# ruff: noqa
-
-import math
+from pathlib import Path
 
-import numpy as np
 import torch
+from torchvision import io
+from torchvision.datasets.folder import IMG_EXTENSIONS
+from torchvision.transforms import v2
 
+from anomalib.data.transforms import MultiRandomChoice
 
-def lerp_np(x, y, w):
-    """Helper function."""
-    return (y - x) * w + x
-
-
-def rand_perlin_2d_octaves_np(shape, res, octaves=1, persistence=0.5):
-    """Generate Perlin noise parameterized by the octaves method. Numpy version."""
-    noise = np.zeros(shape)
-    frequency = 1
-    amplitude = 1
-    for _ in range(octaves):
-        noise += amplitude * generate_perlin_noise_2d(shape, (frequency * res[0], frequency * res[1]))
-        frequency *= 2
-        amplitude *= persistence
-    return noise
 
+def generate_perlin_noise(
```

Comment:
Maybe for a different PR but it would be nice if we could remove inline methods.